### PR TITLE
feat: atomic-product-listing-pager

### PR DESCRIPTION
**Pull request by thudon-coveo**
[COMSDZ-550](https://coveord.atlassian.net/browse/COMSDZ-550)

In this PR:

The `atomic-product-listing-pager` component which is the equivalent to the `atomic-pager` for the `ProductListingEngine`

[COMSDZ-550]: https://coveord.atlassian.net/browse/COMSDZ-550?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/product-listing/atomic-product-listing-pager/atomic-product-listing-pager.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import {SearchStatus, SearchStatusState} from '@coveo/headless';
```
Comment (olamothe) [inline packages/atomic/src/components/product-listing/atomic-product-listing-pager/atomic-product-listing-pager.tsx:1]: This a problematic import: 

By mixing imports from both `@coveo/headless` and `@coveo/headless/product-listing` just below, this defeats the purpose of having different sub packages to have an optimized payload that contains only the necessary code needed to power product listing.

With this import it means you are pulling in a bunch of unrelated code to product listing as part of the final shipped bundle to end users.


The correct approach will be to mimic the job that has been done for `headless-insight-status` (ie: create a `headless-product-listing-status` and re-export that new function from `product-listing.index.ts`

Comment (thudon-coveo) [reply packages/atomic/src/components/product-listing/atomic-product-listing-pager/atomic-product-listing-pager.tsx:1]: Will do, thanks for the explanation.

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/product-listing/atomic-product-listing-pager/atomic-product-listing-pager.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+ * The `atomic-product-listing-pager` provides buttons that allow the end user to navigate through the different result pages.
+ *
+ * @part buttons - The list of the next/previous buttons and page-buttons.
+ * @part page-buttons - The list of page buttons.
+ * @part page-button - The page button.
+ * @part active-page-button - The active page button.
+ * @part previous-button - The previous button.
+ * @part next-button - The next button.
+ * @part previous-button-icon - Icon of the previous button.
+ * @part next-button-icon - Icon of the next button.
```
Comment (olamothe) [inline packages/atomic/src/components/product-listing/atomic-product-listing-pager/atomic-product-listing-pager.tsx:34]: Similar to my comment on your other PR, I would mark this one `@internal` for the time being.

**Review by samuelfullerthomas — APPROVED** [review]

**Comment by olamothe** [comment]
Closing those PR as there was a focus shift regarding those component in the commerce unit.

Re-open if necessary.

**Pull request closed without merging**